src/documentation/io/built-in-google-bigquery.md

```diff
@@ -0,0 +1,753 @@
+%}```
+```py
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/sdks/python/apache_beam/examples/snippets/snippets.py tag:model_bigqueryio_table_spec
+%}```
+
+You can also omit `project_id` and use the `[dataset_id].[table_id]` format. If
+you omit the project ID, Beam uses the default project ID from your
+<span class="language-java">
+  [pipeline options]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/extensions/gcp/options/GcpOptions.html).
+</span>
+<span class="language-py">
+  [pipeline options]({{ site.baseurl }}/documentation/sdks/pydoc/{{ site.release_latest }}/apache_beam.options.pipeline_options.html#apache_beam.options.pipeline_options.GoogleCloudOptions).
+</span>
+
+```java
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/examples/java8/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryTableSpecWithoutProject
+%}```
+```py
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/sdks/python/apache_beam/examples/snippets/snippets.py tag:model_bigqueryio_table_spec_without_project
+%}```
+
+#### Using a TableReference
+
+To specify the target table with a `TableReference`, create a new
+`TableReference` using the three parts of the BigQuery table name.
+
+```java
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/examples/java8/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryTableSpecObject
+%}```
+```py
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/sdks/python/apache_beam/examples/snippets/snippets.py tag:model_bigqueryio_table_spec_object
+%}```
+
+<!-- Java specific -->
+
+{:.language-java}
+The Beam SDK for Java also provides the `parseTableSpec` helper method, which
+constructs a `TableReference` object from a String that contains the
+fully-qualified BigQuery table name. However, the static factory methods for
+BigQueryIO transforms accept the table name as a String and construct a
+`TableReference` object for you.
+
+
+### Table rows
+
+{:.language-java}
+BigQueryIO read and write transforms produce and consume data as either a
+`PCollection` of BigQuery `TableRow` objects, or a `PCollection` of user-defined
+types.
+
+{:.language-py}
+BigQueryIO read and write transforms produce and consume data as a `PCollection`
+of dictionaries, where each element in the `PCollection` represents a single row
+in the table.
+
+### Schemas
+
+When writing to BigQuery, you must supply a table schema for the fields you want
+to write to the target table unless you specify a [create disposition]() of
+`CREATE_NEVER`. [Creating a table schema](#creating-a-table-schema) covers
+schemas in more detail.
+
+
+## Reading from BigQuery
+
+BigQueryIO allows you to read from a BigQuery table, or read specific fields
+using a query string. When you apply a BigQueryIO read transform in batch mode,
```

Comment:
Remove "in batch mode" - Beam doesn't have modes (only bounded or unbounded PCollection's) and the read transform does not have separate bounded/unbounded codepaths (it's not applied to a PCollection, but to the root of the pipeline).

Dataflow runner does have modes, and using BigQueryIO.read() is equally valid in batch and streaming Dataflow runner, and it works the same way.

Reply:
Done